src/secondary_threads/parse_packets.rs

```diff
@@ -45,152 +49,188 @@ pub fn parse_packets(
+                                    .addresses_waiting_resolution
+                                    .entry(address_to_lookup)
+                                    .and_modify(|data_info| {
+                                        data_info.add_packet(
                                             exchanged_bytes,
                                             new_info.traffic_direction,
                                         );
                                     });
-                                }
                             }
-
-                            //increment the packet count for the sniffed app protocol
-                            info_traffic
-                                .app_protocols
-                                .entry(application_protocol)
-                                .and_modify(|data_info| {
-                                    data_info
+                            (_, true) => {
+                                // rDNS already resolved
+                                // update the corresponding host's data info
+                                let host = info_traffic
+                                    .addresses_resolved
+                                    .get(&address_to_lookup)
+                                    .unwrap()
+                                    .1
+                                    .clone();
+                                info_traffic.hosts.entry(host).and_modify(|data_info_host| {
+                                    data_info_host
+                                        .data_info
                                         .add_packet(exchanged_bytes, new_info.traffic_direction);
-                                })
-                                .or_insert(DataInfo::new_with_first_packet(
-                                    exchanged_bytes,
-                                    new_info.traffic_direction,
-                                ));
+                                });
+                            }
                         }
+
+                        //increment the packet count for the sniffed app protocol
+                        info_traffic
+                            .app_protocols
+                            .entry(application_protocol)
+                            .and_modify(|data_info| {
+                                data_info.add_packet(exchanged_bytes, new_info.traffic_direction);
+                            })
+                            .or_insert(DataInfo::new_with_first_packet(
+                                exchanged_bytes,
+                                new_info.traffic_direction,
+                            ));
                     }
                 }
             }
         }
     }
 }
+
+fn get_sniffable_headers<'a>(
+    packet: &'a Packet,
+    link_type: Linktype,
+) -> Result<PacketHeaders<'a>, ReadError> {
+    match link_type {
+        Linktype(12) | Linktype::IPV4 | Linktype::IPV6 => PacketHeaders::from_ip_slice(packet),
```

Comment:
Here I added `LinkType(12)` because doing some tests I discovered that some adapters are tagged with this instead of `IPV4` or `IPV6`.
However a link type of 12 is [not listed anywhere](https://www.tcpdump.org/linktypes.html) so I'm not sure where it's coming from...
Any idea @JulianSchmid?

Reply:
When I have a look at my machine I see some interfaces that are tagged as both "Ethernet" and "12":

```
Found device! Device { name: "ap1", desc: None, addresses: [Address { addr: _, netmask: Some(ffff:ffff:ffff:ffff::), broadcast_addr: None, dst_addr: None }], flags: DeviceFlags { if_flags: UP | RUNNING, connection_status: Disconnected } }
  Main data link: Linktype(1) Ok("EN10MB") Ok("Ethernet")
    Linktype(1) Ok("EN10MB") Ok("Ethernet")
    Linktype(12) Ok("RAW") Ok("Raw IP")
```

So 12 seems to be "RAW IP". But I never did see it as a main data link type only in the list of data links.

I just setup a crate to print all PCAP devices https://github.com/JulianSchmid/print-pcap-devices/tree/v0.1.0 https://crates.io/crates/print-pcap-devices . Maybe have a look what it says there?

Reply:
I already used that script and I obtained similar results.
On Linux TUNs have 12 also as main link type.
![Screenshot 2023-12-31 at 14 00 01](https://github.com/GyulyVGC/sniffnet/assets/100347457/8a2529f8-fe34-423b-938a-b2c2f3ff638c)

I was just wondering why it's not tagged as IPV4 or IPV6 instead, but apparently is just a different way to define the same thing...

Reply:
Ah "TUN" I remember what those are. That are "virtual" network devices that can be used to send & receive network traffic directly on the IP layer. Use-full for VPNs, VMs & so on.

So at the end of the day, yeah from a sniffnets perspective it is the same 😅.

Reply:
Perfect, in fact I tried it and all seems to work properly.

Thank you very much for all the help 🙏 💯